### PR TITLE
Welcome page

### PR DESCRIPTION
**Pull request by kazbaig**
*Issue #, if available:*

*Description of changes:*
Adds welcome page to which the user is redirected if the `welcomePageVisited` flag in localStorage via `vuex` is false or missing and the user is not authenticated.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/web-ui/src/public/Welcome.vue
Comment (alexchirayath) [inline src/web-ui/src/public/Welcome.vue:30]: The Retail Demo Store has certain functionality that does not need sign in such viewing products/similar items , checkiing out as guest or accessing the demo guide. Do we need to add a "Skip this step" button?

Comment (kazbaig) [reply src/web-ui/src/public/Welcome.vue:30]: Definitely makes sense to have that. We've opted to add a "skip this step" button to the roadmap and will make sure to add it before merging into the main branch.

**Review by alexchirayath — APPROVED** [review]
LGTM. Please wait until Jame's review to merge in

File: src/web-ui/src/public/Welcome.vue
Comment (james-jory) [inline src/web-ui/src/public/Welcome.vue:42]: I think we'll want to have this be more dynamic. Most deployments of this project will not be using the reset functionality at all. For example, if I deploy directly from GH. Also, and this is an open question for @alexchirayath, will the reset functionality be unique to the always-on mlsld deployment or something that can be optionally setup for any deployment? Finally, 11pm ET seems kind of early. Should that be configurable and then be dynamically displayed here?

File: src/web-ui/src/public/Welcome.vue
Comment (james-jory) [inline src/web-ui/src/public/Welcome.vue:87]: There are many more services than these used in the project. I know that the intent is to highlight the services that are related to customer experience but maybe we need to set that up better. I'm sure we'll be making some wording/marketing copy sweeps across all of these changes before it's all done so leaving this as a thought for now.

File: src/web-ui/src/config/localStorage/index.js
Comment (james-jory) [inline src/web-ui/src/config/localStorage/index.js:22]: Why not use the existing persisted state mechanism already in place in store.js?

**Review by james-jory — CHANGES_REQUESTED** [review]

File: src/web-ui/src/public/Welcome.vue
Comment (alexchirayath) [reply src/web-ui/src/public/Welcome.vue:42]: Thats a fair point. This probably should not be incorporated into the public demo.
The periodic reset mechanism is built into the deployment architecture (with CodeBuild and Step functions) which is outside of the retail demo store scope.

File: src/web-ui/src/public/Welcome.vue
Comment (kazbaig) [reply src/web-ui/src/public/Welcome.vue:67]: That's fair. We'll be sure to revisit this issue before our deliverable date.

File: src/web-ui/src/config/localStorage/index.js
Comment (kazbaig) [reply src/web-ui/src/config/localStorage/index.js]: Moved this logic to use `vuex`.

File: src/web-ui/src/public/Welcome.vue
Comment (kazbaig) [reply src/web-ui/src/public/Welcome.vue]: Removed the section for now.

**Review by james-jory — APPROVED** [review]

**Pull request merged**
